microprot/scripts/processing.py

```diff
@@ -1,45 +1,155 @@
     """ Extract sequence(s) from a multi-sequence FASTA file.
 
     Parameters
     ----------
     infile : str
         file path to input multi-sequence FASTA file
-    seqidx : int (optional)
-        n-th sequence of the input file to extract
-        (default: 0 for all sequences)
+    identifiers : str or list of str (optional)
+        IDs (names) or indexes (n-th sequence in the file) of the proteins to
+        extract, may be:
+        1) Python list
+        2) comma-separated list
+        3) file path to the list (one ID or index per line)
+        if omitted, all sequences will be extracted
 
-    Returns
-    -------
+    Return
+    ------
     list of skbio Sequence
+        extracted protein sequences
     """
+    l, ids, indexes = [], set(), set()
+    if identifiers:
+        if type(identifiers) is list:  # Python list
+            l = identifiers
+        elif os.path.isfile(identifiers):  # read from a local file
+            with open(identifiers, 'r') as f:
+                l = f.read().splitlines()
+        else:  # comma-separated list
+            l = list(map(str.strip, identifiers.split(',')))
+        for i in l:
+            if i.isdigit():
+                indexes.add(int(i))  # index of this protein in the file
+            else:
+                ids.add(i)  # protein ID (name)
     seqs = []
-    iseq = 0  # current sequence index
-    for seq in io.read(infile, format='fasta'):
-        iseq += 1
-        if 0 < seqidx != iseq:
-            continue
-        seqs.append(seq)
+    for i, seq in enumerate(io.read(infile, format='fasta')):
+        if ids:
+            if seq.metadata['id'] in ids:
+                seqs.append(seq)
+        elif indexes:
+            if i+1 in indexes:  # indexes start with 1, not 0
+                seqs.append(seq)
+        else:
+            seqs.append(seq)
     return seqs
 
 
+def write_sequences(seqs, outfile):
+    """ Write sequence(s) into a multi-sequence FASTA file.
+
+    Parameters
+    ----------
+    seqs : str
+        list of skbio Sequence
+    outfile : str
+        file path to output multi-sequence FASTA file
+    """
+    def outseqs():
+        for seq in seqs:
+            yield seq
+    io.write(outseqs(), format='fasta', into=outfile)
+
+
+def read_representatives(represent):
+    """ Read representative protein IDs from file or server
+
+    Parameters
+    ----------
+    represent : str
+        file path to list of representative protein IDs in XML format, or an
+        integer representing the clustering level by which the list will be
+        downloaded from the PDB server
+
+    Return
+    ------
+    list of str
+        protein IDs
+
+    Raises
+    ------
+    ValueError
+        if neither repfile nor cluster is specified
+    """
+    ids = set()
+    xmldoc = None
+    if os.path.isfile(represent):  # read from a local file
+        xmldoc = minidom.parse(represent)
+    elif represent.isdigit():  # download from the PDB server
```

Comment:
The problem here is that the file to be downloaded is subject to regular update by the server. It is not possible to predict how it will look like. Do you have an idea how to test it?